pkg/result/processors/autogenerated_exclude.go

```diff
@@ -130,11 +130,32 @@ func getDoc(filePath string) (string, error) {
 	scanner.Buffer(make([]byte, initialSize), maxSize)
 
 	var docLines []string
+	inBlockComment := false
 	for scanner.Scan() {
 		line := strings.TrimSpace(scanner.Text())
-		if strings.HasPrefix(line, "//") {
+		if inBlockComment {
```

Comment:
sorry, I don't think it's a good idea to manually parse comments.
Maybe we can use [ast.ParseFile](https://golang.org/pkg/go/parser/#ParseFile) with flag like `PackageClauseOnly` or similar to stop early? I'm sure that performance won't be an issue here if we implement it properly.

Reply:
Thank you for your advice.
I was actually on the fence about using AST, but I'll try to do that.

Reply:
I've changed to stop parsing manually by using `parser.ParseFile()`.
Although `parser.ParseFile()` reads entire file regardless of mode, with `PackageClauseOnly` mode (and `ParseComments` mode), it stops parsing files after import statements, as you pointed out.

Reply:
> The text may appear anywhere in the file.

https://github.com/golang/go/issues/13560#issuecomment-288457920

Are we sure that avoiding possible performance regression has higher priority that compliance with convention?

Reply:
I agree that it looks more important. I guess time still will be <100ms total. We can easily check it: `golangci-lint run -v` shows all processors' times.

Reply:
Do you mean compliance with convention is more important?

Suppose you think so.. performance regression depends on how many issues will be reported on generated files, so I think it's difficult to choose which repository to measure for this purpose. But as you explain in [usage](https://golangci-lint.run/usage/performance/), parsing file seems not so much cost.

> Parsing one source file takes 200 us on average. Parsing of all files in $GOROOT/src takes 2 seconds. Currently we parse each file more than once because it's not the bottleneck.

However, you also write like this:

> We're planning to parse each file only once.

If we check entire file for comments, should we use existing AST instead of parsing file here again?
I haven't confirmed if it's possible yet, I guess we can do so since this function is called on reporting phase.